trillian/integration/ct_integration.go

```diff
@@ -589,6 +593,178 @@ func RunCTIntegrationForLog(cfg *configpb.LogConfig, servers, metricsServers, te
 	return nil
 }
 
+// RunCTLifecycleForLog does a simple log lifecycle test. The log
+// is assumed to be newly created when this test runs. A random number of
+// entries are then submitted to build up a queue. The log is set to
+// DRAINING state and the test checks that all the entries are integrated
+// into the tree and we can verify a consistency proof to the latest entry.
+func RunCTLifecycleForLog(cfg *configpb.LogConfig, servers, metricsServers, adminServer string, testdir string, mmd time.Duration, stats *logStats) error {
+	// Retrieve the test data.
+	testDir := "../testdata"
+	caChain, err := GetChain(testDir, "int-ca.cert")
+	if err != nil {
+		return err
+	}
+	leafChain, err := GetChain(testDir, "leaf01.chain")
+	if err != nil {
+		return err
+	}
+	signer, err := MakeSigner(testDir)
+	if err != nil {
+		return err
+	}
+	leafCert, err := x509.ParseCertificate(leafChain[0].Data)
+	if err != nil {
+		return err
+	}
+	caCert, err := x509.ParseCertificate(caChain[0].Data)
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	pool, err := NewRandomPool(servers, cfg.PublicKey, cfg.Prefix)
+	if err != nil {
+		return fmt.Errorf("failed to create pool: %v", err)
+	}
+	t := testInfo{
+		prefix:         cfg.Prefix,
+		cfg:            cfg,
+		metricsServers: metricsServers,
+		adminServer:    adminServer,
+		stats:          stats,
+		pool:           pool,
+	}
+
+	if err := t.checkStats(); err != nil {
+		return fmt.Errorf("unexpected stats check: %v", err)
```

Comment:
unexpected error from stats check?

Reply:
Was an existing message but I've updated it everywhere.